lib/note-info/reference.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { Fragment, FunctionComponent } from 'react';
+import { connect } from 'react-redux';
+import format from 'date-fns/format';
+
+import actions from '../state/actions';
+import { getNoteReference } from '../utils/get-note-references';
+
+import type * as S from '../state';
+import type * as T from '../types';
+
+type OwnProps = {
+  noteId: T.EntityId;
+};
+
+type StateProps = {
+  reference:
+    | {
+        count: number;
+        noteId: T.EntityId;
+        modificationDate: T.SecondsEpoch;
+        title: string;
+      }
+    | undefined;
+};
+
+type DispatchProps = {
+  openNote: () => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+function formatTimestamp(unixTime: number) {
+  return format(unixTime * 1000, 'MM/dd/yyyy');
+}
+
+export const Reference: FunctionComponent<Props> = ({
+  openNote,
+  reference,
+}) => {
+  if (!reference) {
+    return null;
+  }
+  const formattedDate =
+    reference.modificationDate && formatTimestamp(reference.modificationDate);
+  return (
+    <button className="reference-link" onClick={openNote}>
+      <span className="reference-title note-info-name">{reference.title}</span>
+      <span>
+        {reference.count} References, Last modified {formattedDate}
+      </span>
+    </button>
+  );
+};
+
+const mapStateToProps: S.MapState<StateProps, OwnProps> = (
+  state,
+  { noteId }
+) => ({
+  reference: getNoteReference(state, noteId),
+});
+
+const mapDispatchToProps: S.MapDispatch<DispatchProps, OwnProps> = (
+  dispatch,
+  { noteId }
+) => ({
+  openNote: () => {
+    dispatch(actions.ui.toggleNoteInfo());
```

Comment:
I think the more React-y way to do this would be to add a case to `TOGGLE_NOTE_INFO` that would return false when the `SELECT_NOTE` action is called, unless we can think of any case where we would want the note info panel to remain open (I can't).

Reply:
good call.